src/main/java/edu/cornell/kfs/coa/batch/service/impl/CreateWorkdayOpenAccountsCsvServiceImpl.java

```diff
@@ -0,0 +1,147 @@
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Locale;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+
+import com.opencsv.bean.StatefulBeanToCsv;
+import com.opencsv.bean.StatefulBeanToCsvBuilder;
+import com.opencsv.exceptions.CsvDataTypeMismatchException;
+import com.opencsv.exceptions.CsvRequiredFieldEmptyException;
+
+import edu.cornell.kfs.coa.batch.CuCoaBatchConstants;
+import edu.cornell.kfs.coa.batch.CuCoaBatchConstants.WorkdayOpenAccountDetailDTOCsvColumn;
+import edu.cornell.kfs.coa.batch.businessobject.WorkdayOpenAccountDetailDTO;
+import edu.cornell.kfs.coa.batch.dataaccess.WorkdayOpenAccountDao;
+import edu.cornell.kfs.coa.batch.service.CreateWorkdayOpenAccountsCsvService;
+import edu.cornell.kfs.pmw.batch.PaymentWorksConstants.PaymentWorksUploadFileColumn;
+import edu.cornell.kfs.pmw.batch.businessobject.PaymentWorksVendor;
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.sys.util.EnumConfiguredMappingStrategy;
+
+public class CreateWorkdayOpenAccountsCsvServiceImpl implements CreateWorkdayOpenAccountsCsvService {
+    private static final Logger LOG = LogManager.getLogger();
+    
+    protected String csvOpenAccountsExportDirectory;
+    protected String csvOpenAccountstFileCreationDirectory;
+    protected DateTimeService dateTimeService;
+    protected WorkdayOpenAccountDao workdayOpenAccountDao;
+    
+    @Override
+    public void createWorkdayOpenAccountsCsv() throws IOException {
+        List<WorkdayOpenAccountDetailDTO> details = workdayOpenAccountDao.getWorkdayOpenAccountDetails();
+        if (CollectionUtils.isNotEmpty(details)) {
+            LOG.info("createWorkdayOpenAccountsCsv, number of open account detail lines to write: " + details.size());
+            writeOpenAccountsToCsvFile(details);
+        } else {
+            LOG.error("createWorkdayOpenAccountsCsv, no open accounts were found, this shouldn't happen");
+            throw new RuntimeException("No open accounts were found.");
+        }
+        
+    }
+    
+    private void writeOpenAccountsToCsvFile (List<WorkdayOpenAccountDetailDTO> details) {
+        BufferedWriter bufferedWriter = null;
+        String csvFileName = generateCsvOutputFileName();
+        String fullyQualifiedCreationDirectoryFileName = fullyQualifyFileNameToCreationDirectory(csvFileName);
+        LOG.info("writeOpenAccountsToCsvFile: fullyQualifiedOutputFile = " + fullyQualifiedCreationDirectoryFileName);
+        File outputFile = new File(fullyQualifiedCreationDirectoryFileName);
+        try {
+            OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(fullyQualifiedCreationDirectoryFileName), 
```

Comment:
I would suggest assigning the wrapped FileOutputStream to its own variable, so that it can be explicitly closed afterwards.

Reply:
OK